### PR TITLE
Add support for Pod Topology Spread Constraints

### PR DESCRIPTION
**Pull request by EladDolev**
We'd love having this for increased resiliency when losing an AZ, for both small or large CRDB clusters

### CODE REVIEW

File: cockroachdb/README.md [hunk 1]
```diff
@@ -240,93 +240,97 @@ Note that there are some backward-incompatible changes to SQL features between v
...
+| `statefulset.priorityClassName`                           | [PriorityClassName][4] for StatefulSet Pods                     | `""`                                             |
+| `statefulset.tolerations`                                 | Node taints to tolerate by StatefulSet Pods                     | `[]`                                             |
+| `statefulset.topologySpreadConstraints`                   | [Topology Spread Constraints rules][5] of StatefulSet Pods      | auto                                             |
+| `statefulset.topologySpreadConstraints.maxSkew`           | Degree to which Pods may be unevenly distributed                | `1`                                              |
+| `statefulset.topologySpreadConstraints.topologyKey`       | The key of node labels                                          | `topology.kubernetes.io/zone`                    |
+| `statefulset.topologySpreadConstraints.whenUnsatisfiable` | `ScheduleAnyway`/`DoNotSchedule` for unsatisfiable constraints  | `ScheduleAnyway`                                 |
```
Comment (EladDolev) [inline cockroachdb/README.md:283]: only these were added
the rest of the lines were just aligned

File: cockroachdb/templates/statefulset.yaml [hunk 2]
```diff
@@ -128,6 +128,21 @@ spec:
         {{- end }}
       {{- end }}
     {{- end }}
+    {{- if semverCompare ">=1.16-0" .Capabilities.KubeVersion.Version }}
```
Comment (DuskEagle) [inline cockroachdb/templates/statefulset.yaml:131]: I don't think this check is working as expected, based on the [test failure](https://teamcity.cockroachdb.com/viewLog.html?buildId=2326789&buildTypeId=Cockroach_HelmCharts_ChartsGitHubCi&tab=buildLog&_focus=465). `.Capabilities.KubeVersion.Version` [appears](https://github.com/helm/helm/blob/0ca95341c19fcaf9cecdf50caef1693568835a84/pkg/chartutil/capabilities.go#L48) to be a [k8s.io/apimachinery/pkg/version#Info](https://godoc.org/k8s.io/apimachinery/pkg/version#Info), which does not have a `Version` field.

**Review by DuskEagle — CHANGES_REQUESTED** [review]

File: cockroachdb/templates/statefulset.yaml (hunk 2 above)
Comment (EladDolev) [reply cockroachdb/templates/statefulset.yaml]: changed to `.Capabilities.KubeVersion.GitVersion` which is deprecated on Helm v3 but still works :-)

**Review by DuskEagle — APPROVED** [review]
LGTM. I've updated the Chart.yaml version on this PR as well.

**Pull request merged**